tuf/download.py

```diff
@@ -382,16 +379,8 @@ def _download_fixed_amount_of_data(response, temp_file, required_length):
         break
 
   except urllib3.exceptions.ReadTimeoutError as e:
-    # Whatever happens, make sure that we always close the connection.
-    response.close()
     raise tuf.exceptions.SlowRetrievalError(str(e))
 
-  except:
-    # Whatever happens, make sure that we always close the connection.
-    response.close()
-    raise
-
-  response.close()
```

Comment:
Nice. This should probably have been a `try/except/finally` to ensure the response was always closed.

Reply:
Yes, that was definitely a potential future bug